javascript-carousel/main.js

```diff
@@ -0,0 +1,184 @@
+  $dot5.className = 'dotView far fa-circle fa-lg';
+
+  var newView = '1';
+
+  for (var i = 0; i < $allView.length; i++) {
+    if ($allView[i].getAttribute('data-view') === newView) {
+      $allView[i].className = 'view';
+    } else {
+      $allView[i].className = 'view hidden';
+    }
+  }
+
+  currentView = newView;
+  clearInterval(timerId);
+  timerId = setInterval(handleRight, 3000);
+}
+
+function handleDot2(event) {
+  $dot1.className = 'dotView far fa-circle fa-lg';
+  $dot2.className = 'dotView fas fa-circle fa-lg';
+  $dot3.className = 'dotView far fa-circle fa-lg';
+  $dot4.className = 'dotView far fa-circle fa-lg';
+  $dot5.className = 'dotView far fa-circle fa-lg';
+
+  var newView = '2';
+
+  for (var i = 0; i < $allView.length; i++) {
+    if ($allView[i].getAttribute('data-view') === newView) {
+      $allView[i].className = 'view';
+    } else {
+      $allView[i].className = 'view hidden';
+    }
+  }
+
+  currentView = newView; clearInterval(timerId);
+  timerId = setInterval(handleRight, 3000);
+}
+
+function handleDot3(event) {
+  $dot1.className = 'dotView far fa-circle fa-lg';
+  $dot2.className = 'dotView far fa-circle fa-lg';
+  $dot3.className = 'dotView fas fa-circle fa-lg';
+  $dot4.className = 'dotView far fa-circle fa-lg';
+  $dot5.className = 'dotView far fa-circle fa-lg';
+
+  var newView = '3';
+
+  for (var i = 0; i < $allView.length; i++) {
+    if ($allView[i].getAttribute('data-view') === newView) {
+      $allView[i].className = 'view';
+    } else {
+      $allView[i].className = 'view hidden';
+    }
+  }
+
+  currentView = newView;
+  clearInterval(timerId);
+  timerId = setInterval(handleRight, 3000);
+}
+
+function handleDot4(event) {
+  $dot1.className = 'dotView far fa-circle fa-lg';
+  $dot2.className = 'dotView far fa-circle fa-lg';
+  $dot3.className = 'dotView far fa-circle fa-lg';
+  $dot4.className = 'dotView fas fa-circle fa-lg';
+  $dot5.className = 'dotView far fa-circle fa-lg';
+
+  var newView = '4';
+
+  for (var i = 0; i < $allView.length; i++) {
+    if ($allView[i].getAttribute('data-view') === newView) {
+      $allView[i].className = 'view';
+    } else {
+      $allView[i].className = 'view hidden';
+    }
+  }
+
+  currentView = newView;
+  clearInterval(timerId);
+  timerId = setInterval(handleRight, 3000);
+}
+
+function handleDot5(event) {
+  $dot1.className = 'dotView far fa-circle fa-lg';
+  $dot2.className = 'dotView far fa-circle fa-lg';
+  $dot3.className = 'dotView far fa-circle fa-lg';
+  $dot4.className = 'dotView far fa-circle fa-lg';
+  $dot5.className = 'dotView fas fa-circle fa-lg';
+
+  var newView = '5';
+
+  for (var i = 0; i < $allView.length; i++) {
+    if ($allView[i].getAttribute('data-view') === newView) {
+      $allView[i].className = 'view';
+    } else {
+      $allView[i].className = 'view hidden';
+    }
+  }
+
+  currentView = newView;
+  clearInterval(timerId);
+  timerId = setInterval(handleRight, 3000);
+}
```

Comment:
LOTS of repetition here. This can absolutely be combined somehow. If you need help coming up with ideas on how, let me know 👍